pace-util/pace/util/report.py

```diff
@@ -0,0 +1,114 @@
+import copy
+import json
+from datetime import datetime
+from typing import Any, Dict, List
+
+import numpy as np
+
+from pace.util.mpi import MPI
+
+
+def set_experiment_info(
+    experiment_name: str, time_step: int, backend: str, git_hash: str
+) -> Dict[str, Any]:
+    experiment: Dict[str, Any] = {}
+    now = datetime.now()
+    dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
+    experiment["setup"] = {}
```

Comment:
There's a lot of use of nested dictionaries here - this might be easier to work with and harder to make bugs on if you turn this into a dataclass structure.

Reply:
Done!